src/transformers/models/data2vec/modeling_data2vec_text.py

```diff
@@ -816,7 +823,18 @@ def forward(
             encoder_hidden_shape = (encoder_batch_size, encoder_sequence_length)
             if encoder_attention_mask is None:
                 encoder_attention_mask = torch.ones(encoder_hidden_shape, device=device)
-            encoder_extended_attention_mask = self.invert_attention_mask(encoder_attention_mask)
+
+            if self._use_sdpa and head_mask is None and not output_attentions:
+                # Expand the attention mask for SDPA.
+                # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+                encoder_extended_attention_mask = _prepare_4d_causal_attention_mask_for_sdpa(
+                    encoder_attention_mask,
+                    input_shape,
+                    input_ids_or_embeds,
+                    past_key_values_length,
+                )
+            else:
+                encoder_extended_attention_mask = self.invert_attention_mask(encoder_attention_mask)
```

Comment:
I don't think `_prepare_4d_causal_attention_mask_for_sdpa` & `invert_attention_mask` do the same thing.

Maybe you rather need `_prepare_4d_attention_mask_for_sdpa`

Reply:
Good catch. I've fixed it so that we only use the causal one for decoder attention masks and only when the model is a decoder